---
title: "Statin data"
output: html_notebook
---

This is the statin dataset. Upon loading the dataset, we wil clean the data by:  
1) converting NAs to zeroes  
2) removing other timepoint readings except 12h measurements  
3) convert certain grouping variables to factor classes

```{r message=FALSE}
## load raw dataset
df <- read.csv(file = "Patient Clinical Characteristics (No Dup_Trimmed Genetics).csv")

#### data cleaning ####
## remove NA
df[is.na(df)] <- 0
local({
  for(i in c("5h", "6h", "7h", "8h", "9h", "10h", "11h", "13h", "14h")){
    df <<- df[,-grep(i,colnames(df))]
  }
})
local({
  to_convert <- c("Site_Code", "GenderCode", "BMICode", "RaceCode", "StatinCode", 
                  "StatinDoseCode", "MI", "Renal_Problems", "Liver_Problems", "Hypertension", 
                  "Diabetes_Mellitus", "Hypercholesterolemia", "Myalgia", "Myalgia_Score", "Blood_Thinner",
                  "Glucose_lowering", "Cholesterol_Lowering", "Heart_protective", "BP_lowering")
  conv_to_factor <- function(datafram, c){
    datafram[, c] <- as.factor(datafram[, c])
    return(datafram)
  }
  for(i in to_convert){
    df <<- conv_to_factor(df, i)
  }
  df$Subject_ID <<- as.character(df$Subject_ID)
})

## identify which columns are classed wrong (skipped since I found alr)
type_df <- list()
local({
  data_headers <- colnames(df)
  types <- character()
  for(i in seq_along(data_headers)){
    type_i <- class(df[, i])
    types <- c(types, type_i)
  }
  type_df <<- list(header = data_headers, type = types)
})
type_df <- as.data.frame(type_df)
f_groupings <- as.character(type_df[type_df$type == "factor",1])
```

```{r}
print(as.character(type_df$header[type_df$type == "factor"]))
```
These are the variables we can group our data by. We can see patient demographics, clinical information and SNP genotypes.

For comparisons involving hsCRP levels (or any other COBAS readings), we will need to subset the data to those that actually had their CRP levels measured. Here are the measurements of the 89 patients.
```{r}
w.hsCRP <- df[df$COBAS_hsCRP > 0,]
qnt <- quantile(w.hsCRP$COBAS_hsCRP, probs = c(0.2,0.8))
hi.crp <- w.hsCRP[w.hsCRP$COBAS_hsCRP > qnt[[2]],]
lo.crp <- w.hsCRP[w.hsCRP$COBAS_hsCRP < qnt[[1]],]
boxplot(w.hsCRP$COBAS_hsCRP, horizontal = TRUE)
boxplot(w.hsCRP$COBAS_hsCRP, horizontal = TRUE, log = "x")
summary(w.hsCRP$COBAS_hsCRP)
```

**Patient Selection**  
For the 250 targeted sequencing candidate selection, we will subset out  
1) Only patients prescribed a 40 dose (to control for confounding due to varying doses)  
2) Lower and upper quartiles of patients by drug metabolite level  
3) Upper quartile of patients by C-reactive Protein level  
4) Myalgia symptoms (score of 4 or higher)

*Here we've used exclusively patients prescribed atorvastatin, as they made up a larger population and had more reliable data.

```{r}
## use patients with dose 40
dose40_df <- df[df$StatinDose == 40,]
atv_pat <- dose40_df[dose40_df$StatinType == "Atorvastatin",]
x20hatv_qnt <- quantile(atv_pat$X2OHATV12h, probs = c(0.25, 0.75))
x40hatv_qnt <- quantile(atv_pat$X4OHATV12h, probs = c(0.25, 0.75))
atv_qnt <- quantile(atv_pat$ATVCS12h, probs = c(0.25, 0.75))
both <- atv_pat[atv_pat$SIMACID12h > 0,c("Subject_ID", "ATVCS12h", "X2OHATV12h", "X4OHATV12h", "SIM12h", "SIMACID12h")]
## patients with low drug levels
low_2ohatv <- atv_pat[atv_pat$X2OHATV12h <= x20hatv_qnt[[1]],]
low_4ohatv <- atv_pat[atv_pat$X4OHATV12h <= x40hatv_qnt[[1]],]
low_atv <- atv_pat[atv_pat$ATVCS12h <= atv_qnt[[1]],]
length(low_2ohatv$Subject_ID[low_2ohatv$Subject_ID == low_4ohatv$Subject_ID])
low_20h_4oh <- intersect(low_4ohatv$Subject_ID, low_2ohatv$Subject_ID)
low_20h_atv <- intersect(low_2ohatv$Subject_ID, low_atv$Subject_ID)
low_40h_atv <- intersect(low_4ohatv$Subject_ID, low_atv$Subject_ID)
low_all_atv <- intersect(low_20h_4oh, low_20h_atv)
low_atv_total <- atv_pat[atv_pat$Subject_ID %in% low_all_atv,]
```

```{r}
nrow(dose40_df)
nrow(atv_pat)
nrow(low_atv_total)
print(both)
```

![Each group yielded 157 subjects, with high overlap. 130 subjects were identified with low measurements for atorvastatin and its metabolites](Venn_2set_simple.png)

As we can see, there are 626 patients prescribed a 40mg dose of atorvastatin, of which 130 are in the bottom quantile in measurements for plasma levels of atorvastatin and its metabolites. Also of note is that four subjects were identified to have readings for simvastatin, with one of them missing values for atorvastatin. This is likely a data entry error.


